[BUG] Error: Plugin did not respond │ │ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.

### 🐛 What happened?

When try to create a lakehouse, I'm facing this issue
version 1.0.0

The lakehouse has been created in the workspace, but the SQL enpoint + SM still pending.

--------------------->>>>


Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-fabric_v1.0.0.exe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x10 pc=0x1246739]

goroutine 23 [running]:
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils.(*ErrorHandler).processFabricError(0x15d3380?, {0x1a14c30, 0xc0000c31f0}, 0xc0001c21b0, {0x1a05160, 0x2449ed0}, {0x1854db6, 0xe}, {0x185ffc7, 0x17})
        github.com/microsoft/terraform-provider-fabric/internal/pkg/utils/errors.go:159 +0x319
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils.(*ErrorHandler).processError(0xc0004e8b20, {0x1a14c30, 0xc0000c31f0}, {0x1a057a0, 0xc0001c21b0}, {0x1a05160, 0x2449ed0}, {0x1854db6, 0xe}, {0x185ffc7, ...})
        github.com/microsoft/terraform-provider-fabric/internal/pkg/utils/errors.go:134 +0x38f
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils.(*ErrorHandler).GetDiagsFromError(0xc000431c02?, {0x1a14c30, 0xc0000c31f0}, {0x1a057a0, 0xc0001c21b0}, {0x184a6b1?, 0xc0000c2f02?}, 
{0x1a05160?, 0x2449ed0?})
        github.com/microsoft/terraform-provider-fabric/internal/pkg/utils/errors.go:89 +0x1d5
github.com/microsoft/terraform-provider-fabric/internal/pkg/utils.GetDiagsFromError(...)
        github.com/microsoft/terraform-provider-fabric/internal/pkg/utils/errors.go:307
github.com/microsoft/terraform-provider-fabric/internal/pkg/fabricitem.(*ResourceFabricItemConfigProperties[...]).get(0x1a3ab80, {0x1a14c30, 0xc0000c31f0}, 0xc0003c8a50)
        github.com/microsoft/terraform-provider-fabric/internal/pkg/fabricitem/resource_item_config_properties.go:363 +0x213
github.com/microsoft/terraform-provider-fabric/internal/pkg/fabricitem.(*ResourceFabricItemConfigProperties[...]).Create(0x1a3ab80, {0x1a14b88, 0xc0004dad20}, {{{{0x1a22e70, 0xc00015e720}, {0x1699da0, 0xc00015e120}}, {0x1a26498, 0xc00012efa0}}, {{{0x1a22e70, ...}, ...}, ...}, ...}, ...)
        github.com/microsoft/terraform-provider-fabric/internal/pkg/fabricitem/resource_item_config_properties.go:139 +0x905
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0xc0001b3d48, {0x1a14b88, 0xc0004dad20}, 0xc0004e9500, 0xc0004e94d8)
        github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_createresource.go:101 +0x577
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0xc0001b3d48, {0x1a14b88, 0xc0004dad20}, 0xc0002144b0, 0xc0004e95f0)
        github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/fwserver/server_applyresourcechange.go:57 +0x4aa
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0xc0001b3d48, {0x1a14b88?, 0xc0004dac30?}, 0xc0002143c0)
        github.com/hashicorp/terraform-plugin-framework@v1.14.1/internal/proto6server/server_applyresourcechange.go:55 +0x385
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc0002319a0, {0x1a14b88?, 0xc0004da120?}, 0xc0003904d0)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/tf6server/server.go:866 +0x3b9
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x181a1e0, 0xc0002319a0}, {0x1a14b88, 0xc0004da120}, 0xc0004d2000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:611 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000234200, {0x1a14b88, 0xc000394150}, 0xc000108060, 0xc000303bf0, 0x2475e18, 0x0)
        google.golang.org/grpc@v1.69.4/server.go:1392 +0xfbe
google.golang.org/grpc.(*Server).handleStream(0xc000234200, {0x1a15d60, 0xc000402000}, 0xc000108060)
        google.golang.org/grpc@v1.69.4/server.go:1802 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.69.4/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 21
        google.golang.org/grpc@v1.69.4/server.go:1041 +0x11d

Error: The terraform-provider-fabric_v1.0.0.exe plugin crashed!

### 🔬 How to reproduce?

terraform apply

### 🏗️ Code Sample / Log

resource "fabric_lakehouse" "LH_BRONZE_Terraform6_dummy" {
  display_name = "LH_BRONZE_Terraform6"
  workspace_id = "1ba08eee-0035-45fe-8637-0bcd3e126a15"
  timeouts = {
    create = "2h"
  }
}


### 📷 Screenshots

_No response_

### 📈 Expected behavior

The LH created completed

### 🌌 Environment (Provider Version)

1.0.0

### 🌌 Environment (Terraform Version)

Terraform v1.11.3

### 🌌 Environment (OS)

Linux

### 📎 Additional context

_No response_

### 🔰 Code of Conduct

- [x] I agree to follow this project's Code of Conduct.

--- a/.changes/unreleased/fixed-20250409-105232.yaml
+++ b/.changes/unreleased/fixed-20250409-105232.yaml
@@ -1,0 +1,5 @@
+kind: fixed
+body: Nil pointer dereference on Fabric Error Responses when RawResponse is nil.
+time: 2025-04-09T10:52:32.927656615Z
+custom:
+    Issue: "362"

--- a/internal/pkg/utils/errors.go
+++ b/internal/pkg/utils/errors.go
@@ -156,7 +156,7 @@ func (h *ErrorHandler) processFabricError(
 	})
 
 	// Handle special case for error identity check or not found errors
-	if (errIs != nil && errors.Is(errResp, errIs)) || (errIs != nil && errResp.RawResponse.StatusCode == http.StatusNotFound) {
+	if errIs != nil && (errors.Is(errResp, errIs) || (errResp.RawResponse != nil && errResp.RawResponse.StatusCode == http.StatusNotFound)) {
 		return errIs.Error(), errIs.Error()
 	}
 

--- a/internal/pkg/utils/errors_test.go
+++ b/internal/pkg/utils/errors_test.go
@@ -427,16 +427,49 @@ func TestUnit_GetDiagsFromError_FabricError(t *testing.T) {
 	handler := utils.NewErrorHandler()
 	ctx := t.Context()
 
-	fabricErr := createFabricError(t, 400, "InvalidParameter", "The parameter is invalid", "req-123")
+	tests := []struct {
+		name         string
+		fabricErr    *fabcore.ResponseError
+		wantContains []string
+	}{
+		{
+			name:      "normal error response",
+			fabricErr: createFabricError(t, 400, "InvalidParameter", "The parameter is invalid", "req-123"),
+			wantContains: []string{
+				"InvalidParameter",
+				"The parameter is invalid",
+				"req-123",
+			},
+		},
+		{
+			name: "nil RawResponse & nil ErrorResponse.RequestID",
+			fabricErr: &fabcore.ResponseError{
+				StatusCode: 400,
+				ErrorResponse: &fabcore.ErrorResponse{
+					ErrorCode: azto.Ptr("InvalidParameter"),
+					Message:   azto.Ptr("The parameter is invalid"),
+				},
+			},
+			wantContains: []string{
+				"InvalidParameter",
+				"The parameter is invalid",
+			},
+		},
+	}
 
-	diags := handler.GetDiagsFromError(ctx, fabricErr, utils.OperationCreate, nil)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			diags := handler.GetDiagsFromError(ctx, tt.fabricErr, utils.OperationCreate, nil)
 
-	assert.True(t, diags.HasError())
-	diagErrs := diags.Errors()
-	assert.Len(t, diagErrs, 1)
-	assert.Contains(t, diagErrs[0].Detail(), "InvalidParameter")
-	assert.Contains(t, diagErrs[0].Detail(), "The parameter is invalid")
-	assert.Contains(t, diagErrs[0].Detail(), "req-123")
+			assert.True(t, diags.HasError())
+			diagErrs := diags.Errors()
+			assert.Len(t, diagErrs, 1)
+
+			for _, want := range tt.wantContains {
+				assert.Contains(t, diagErrs[0].Detail(), want)
+			}
+		})
+	}
 }
 
 func TestUnit_GetDiagsFromError_AuthFailedError(t *testing.T) {

